sdk/loganalytics/azure-resourcemanager-loganalytics/src/test/java/com/azure/resourcemanager/loganalytics/LogAnalyticsManagerTests.java

```diff
@@ -0,0 +1,101 @@
+import com.azure.resourcemanager.loganalytics.models.*;
+import com.azure.resourcemanager.resources.ResourceManager;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Random;
+
+public class LogAnalyticsManagerTests extends TestBase {
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.US_EAST;
+    private String resourceGroupName = "rg" + randomPadding();
+    private LogAnalyticsManager logAnalyticsManager;
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        logAnalyticsManager = LogAnalyticsManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups()
+                .define(resourceGroupName)
+                .withRegion(REGION)
+                .create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testCreateWorkspace() {
+        Workspace workspace = null;
+        try {
+            String spaceName = "space" + randomPadding();
+            // @embedStart
+            workspace = logAnalyticsManager.workspaces()
+                .define(spaceName)
+                .withRegion(REGION)
+                .withExistingResourceGroup(resourceGroupName)
+                .withSku(new WorkspaceSku().withName(WorkspaceSkuNameEnum.PER_GB2018))
+                .withFeatures(new WorkspaceFeatures().withEnableLogAccessUsingOnlyResourcePermissions(true))
+                .withWorkspaceCapping(new WorkspaceCapping().withDailyQuotaGb(-1D))
+                .withRetentionInDays(30)
+                .withPublicNetworkAccessForIngestion(com.azure.resourcemanager.loganalytics.models.PublicNetworkAccessType.ENABLED)
+                .withPublicNetworkAccessForQuery(com.azure.resourcemanager.loganalytics.models.PublicNetworkAccessType.ENABLED)
```

Comment:
Could we disable it? We prefer not to have public access for test resource, unless necessary.

Reply:
Fixed to be the most streamlined parameters in the new version.